pkg/kubectl/cmd/util/factory.go

```diff
@@ -250,17 +250,23 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		// have been dynamically added to the apiserver
 		Object: func(discoverDynamicAPIs bool) (meta.RESTMapper, runtime.ObjectTyper) {
 			cfg, err := clientConfig.ClientConfig()
-			CheckErr(err)
+			if err != nil {
+				CheckErr(fmt.Errorf("failed to get client config: %v", err))
```

Comment:
How about plumbing the prefix to the `checkErr` function, then you won't need the `err != nil` checks.